admin_guide/allocating_node_resources.adoc

```diff
@@ -155,6 +164,107 @@ $ curl <certificate details> https://<master>/api/v1/nodes/cluster.node22/proxy/
+<3> A comma-delimited list of scopes for where the node should enforce node
+allocatable constraints.  Valid values are "pods", "system-reserved", and "kube-reserved".
+The default is "pods" and is not recommended to change at this time.
+====
+
+Optionally, the node can be made to enforce kube-reserved and system-reserved by
+specifying those tokens in the enforce-node-allocatable flag.  If specified, the
+corresponding --kube-reserved-cgroup or --system-reserved-cgroup needs to be provided.
+In future releases, the node and container runtime will be packaged in a common cgroup
+separate from `system.slice`.  Until that time, it is not encouraged for users to
+change the default value of enforce-node-allocatable flag.
+
+System daemons are expected to be treated similar to Guaranteed pods. System daemons
+can burst within their bounding control groups and this behavior needs to be managed
+as part of cluster deployments.  Enforcing system-reserved reservations
+can lead to critical system services being CPU starved or OOM killed on the node. The
+recommendation is to enforce system-reserved only if operators have profiled their nodes
+exhaustively to come up with precise estimates and are confident in their ability to 
+recover if any process in that group is oom_killed.
+
+As a result, it is strongly recommended that users only enforce node allocatable for
+"pods" by default, and set aside appropriate reservations for system daemons to maintain
+overall node reliability.
+
+[[node-enforcement]]
+== Eviction Thresholds
+
+Memory pressure at the node level leads to system OOMs which affects the entire
+node and all pods running on it.  Nodes can go offline temporarily until memory has
+been reclaimed. To avoid (or reduce the probability of) system OOMs the node
+provides xref:../admin_guide/out_of_resource_handling.adoc[Out Of Resource Handling].
+
+By reserving some memory via the `*--eviction-hard*` flag, the node attempts to evict
+pods whenever memory availability on the node drops below the reserved value.
+Hypothetically, if system daemons did not exist on a node, pods cannot use more than
+`capacity - eviction-hard`. For this reason, resources reserved for evictions are not
+available for pods.
+
+Here is an example to illustrate the impact of node allocatable for memory:
+
+Node capacity is `32Gi`, kube-reserved is `2Gi`, system-reserved is `1Gi`, eviction-hard is set to `<100Mi`.
+
+For this node, the effective node allocatable value is `28.9Gi` only; i.e. if the node
+and system components use up all their reservation, the memory available for pods is `28.9Gi`,
+and kubelet will evict pods when it exceeds this usage.
+
+If we enforce node allocatable (`28.9Gi`) via top level cgroups, then pods can never exceeds `28.9Gi`
+in which case evictions would not be performed unless kernel memory consumption is above `100Mi`.
+
+In order to support evictions and avoid memcg OOM kills for pods, the node sets the top level cgroup
```

Comment:
s/memcg OOM killls for pods/pods being OOM killed inside the kubepods cgroup/

explaining the kubepods cgroup could be worth it so we don't have to use different, confusing language each time we are referring to it.  "pod-level cgroup" already means something else so...